### PR TITLE
Enable IREE_TARGET_BACKEND_WEBGPU in build scripts for CI coverage.

### PR DESCRIPTION
**Pull request by ScottTodd**
Split from https://github.com/iree-org/iree/pull/10134 - this starts building the WebGPU compiler target backend and runs the tests added in that PR.

### CODE REVIEW

**Comment by ScottTodd** [comment]
Besides [an ASan flake](https://github.com/iree-org/iree/actions/runs/3316089449/jobs/5477477103), this seems to be working (not running out of disk space) now, probably thanks to the stop-gap in https://github.com/iree-org/iree/pull/10754.

File: build_tools/cmake/build_all.sh [hunk 1]
```diff
@@ -54,6 +54,10 @@ declare -a CMAKE_ARGS=(
   "-DIREE_HAL_DRIVER_CUDA=ON"
   "-DIREE_TARGET_BACKEND_CUDA=ON"
 
+  # Enable WebGPU compiler builds and tests. All deps get fetched as needed,
+  # but some of the deps are too large to enable by default for all developers.
+  "-DIREE_TARGET_BACKEND_WEBGPU=ON"
```
Comment (pzread) [inline build_tools/cmake/build_all.sh:59]: Do we want to also enable this in `build_and_test_asan.sh` and `build_and_test_tsan.sh`?

**Review by pzread — APPROVED** [review]

File: build_tools/cmake/build_all.sh (hunk 1 above)
Comment (ScottTodd) [reply build_tools/cmake/build_all.sh:59]: Oo I think that's a good idea. I'll try it.

Comment (ScottTodd) [reply build_tools/cmake/build_all.sh:59]: Hmm... the ASan build (not even getting to tests) fails with

```
_deps/tint-build/src/tint/libtint_diagnostic_utils.a(source.cc.o): In function `tint::(anonymous namespace)::ParseLineBreak(std::basic_string_view<char, std::char_traits<char> >, unsigned long, bool*, unsigned long*)':
/work/build-asan/_deps/tint-src/src/tint/source.cc:31: undefined reference to `tint::text::utf8::Decode(unsigned char const*, unsigned long)'
/work/build-asan/_deps/tint-src/src/tint/source.cc:58: undefined reference to `tint::text::utf8::Decode(unsigned char const*, unsigned long)'
```

that happens even when I set `-DTINT_ENABLE_ASAN=ON`, and I'm not sure how much it's worth debugging further.

**Pull request merged**